### DISCUSSION TITLE
Mapping protobuf generated classes to Kotlin data classes

### DISCUSSION THREAD

**Discussion by rrva** [mapstruct/mapstruct#2707] [Q&A]
I have a project where I want to map protobuf generated classes to manually implemented Kotlin data classes.

If the field names and type names correspond 1:1 (the protoc schema is originally generated from the target Kotlin code), what is a good way to do it that involves as little extra code as possible?

**Comment by filiphr** [comment]
I am not quite sure that I understand the question. You can always define your own `@Mapper` classes and provide `@Mapping` for the field names that do not match
